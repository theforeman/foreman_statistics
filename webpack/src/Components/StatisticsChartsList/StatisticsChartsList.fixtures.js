export const statisticsData = {
  operatingsystem: {
    id: 'operatingsystem',
    title: 'OS Distribution',
    url: 'statistics/operatingsystem',
    search: '/hosts?search=os_title=~VAL~',
  },
  architecture: {
    id: 'architecture',
    title: 'Architecture Distribution',
    url: 'statistics/architecture',
    search: '/hosts?search=facts.architecture=~VAL~',
  },
};

export const statisticsMeta = [
  statisticsData.operatingsystem,
  statisticsData.architecture,
];
