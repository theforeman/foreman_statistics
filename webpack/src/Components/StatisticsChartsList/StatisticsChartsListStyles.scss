.statistics-charts-list-root {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  padding-top: 20px;
  .chart-box {
    height: 312px;
    width: 280px;
    display: -ms-flexbox;
    display: flex;
    -ms-flex-direction: column;
    flex-direction: column;
    min-height: 312px;

    .card-pf-heading {
      -ms-flex: 0 0 30px;
      flex: 0 0 30px;
    }

    .card-pf-body {
      display: -ms-flexbox;
      display: flex;
      -ms-flex: 1;
      flex: 1;
      margin: 0;
    }
  }
}
