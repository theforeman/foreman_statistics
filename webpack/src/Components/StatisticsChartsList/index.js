import React from 'react';
import PropTypes from 'prop-types';

import { translate as __ } from 'foremanReact/common/I18n';
import ConnectedChartBox from 'foremanReact/components/ChartBox';
import './StatisticsChartsListStyles.scss';

const StatisticsChartsList = ({ data }) => {
  const chartBoxes = Object.values(data).map(chart => (
    <ConnectedChartBox
      key={chart.id}
      type="donut"
      chart={chart}
      noDataMsg={__('No data available')}
      tip={__('Expand the chart')}
    />
  ));

  return (
    <div className="statistics-charts-list-root cards-pf">{chartBoxes}</div>
  );
};

StatisticsChartsList.propTypes = {
  data: PropTypes.oneOfType([PropTypes.array, PropTypes.object]),
};

StatisticsChartsList.defaultProps = {
  data: [],
};

export default StatisticsChartsList;
