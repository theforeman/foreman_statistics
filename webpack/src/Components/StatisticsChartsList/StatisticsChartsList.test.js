import { testComponentSnapshotsWithFixtures } from '@theforeman/test';

import StatisticsChartsList from './';
import { statisticsData } from './StatisticsChartsList.fixtures';

const fixtures = {
  'should render no panels for empty data': {
    data: {},
  },
  'should render two panels for fixtures data': {
    data: statisticsData,
  },
};

describe('StatisticsChartsList', () => {
  describe('rendering', () =>
    testComponentSnapshotsWithFixtures(StatisticsChartsList, fixtures));
});
