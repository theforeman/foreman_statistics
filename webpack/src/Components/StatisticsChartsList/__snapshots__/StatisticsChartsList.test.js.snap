// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatisticsChartsList rendering should render no panels for empty data 1`] = `
<div
  className="statistics-charts-list-root cards-pf"
/>
`;

exports[`StatisticsChartsList rendering should render two panels for fixtures data 1`] = `
<div
  className="statistics-charts-list-root cards-pf"
>
  <ConnectedChartBox
    chart={
      Object {
        "id": "operatingsystem",
        "search": "/hosts?search=os_title=~VAL~",
        "title": "OS Distribution",
        "url": "statistics/operatingsystem",
      }
    }
    className=""
    config="regular"
    errorText=""
    key="operatingsystem"
    noDataMsg="No data available"
    tip="Expand the chart"
    type="donut"
  />
  <ConnectedChartBox
    chart={
      Object {
        "id": "architecture",
        "search": "/hosts?search=facts.architecture=~VAL~",
        "title": "Architecture Distribution",
        "url": "statistics/architecture",
      }
    }
    className=""
    config="regular"
    errorText=""
    key="architecture"
    noDataMsg="No data available"
    tip="Expand the chart"
    type="donut"
  />
</div>
`;
